auth/user_mgt.go

```diff
@@ -84,6 +86,217 @@ type ExportedUserRecord struct {
+		}
+	}
+	return info, nil
+}
+
+// UID setter. This field is required.
+func (u *UserToImport) UID(uid string) *UserToImport {
+	u.userInfo().LocalId = uid
+	return u
+}
+
+// Email setter.
+func (u *UserToImport) Email(email string) *UserToImport {
+	u.userInfo().Email = email
+	return u
+}
+
+// DisplayName setter.
+func (u *UserToImport) DisplayName(displayName string) *UserToImport {
+	u.userInfo().DisplayName = displayName
+	return u
+}
+
+// PhotoURL setter.
+func (u *UserToImport) PhotoURL(url string) *UserToImport {
+	u.userInfo().PhotoUrl = url
+	return u
+}
+
+// PhoneNumber setter.
+func (u *UserToImport) PhoneNumber(phoneNumber string) *UserToImport {
+	u.userInfo().PhoneNumber = phoneNumber
+	return u
+}
+
+// Metadata setter.
+func (u *UserToImport) Metadata(metadata *UserMetadata) *UserToImport {
+	info := u.userInfo()
+	info.CreatedAt = metadata.CreationTimestamp
+	info.LastLoginAt = metadata.LastLogInTimestamp
+	return u
+}
+
+// ProviderData setter.
+func (u *UserToImport) ProviderData(providers []*UserProvider) *UserToImport {
+	var providerUserInfo []*identitytoolkit.UserInfoProviderUserInfo
+	for _, p := range providers {
+		providerUserInfo = append(providerUserInfo, &identitytoolkit.UserInfoProviderUserInfo{
+			ProviderId:  p.ProviderID,
+			RawId:       p.UID,
+			Email:       p.Email,
+			DisplayName: p.DisplayName,
+			PhotoUrl:    p.PhotoURL,
+		})
+	}
+	u.userInfo().ProviderUserInfo = providerUserInfo
+	return u
+}
+
+// CustomClaims setter.
+func (u *UserToImport) CustomClaims(claims map[string]interface{}) *UserToImport {
+	u.claims = claims
+	return u
+}
+
+// Disabled setter.
+func (u *UserToImport) Disabled(disabled bool) *UserToImport {
+	info := u.userInfo()
+	info.Disabled = disabled
+	if !disabled {
+		info.ForceSendFields = append(info.ForceSendFields, "Disabled")
+	}
+	return u
+}
+
+// EmailVerified setter.
+func (u *UserToImport) EmailVerified(emailVerified bool) *UserToImport {
+	info := u.userInfo()
+	info.EmailVerified = emailVerified
+	if !emailVerified {
+		info.ForceSendFields = append(info.ForceSendFields, "EmailVerified")
+	}
+	return u
+}
+
+// PasswordHash setter. When set a UserImportHash must be specified as an option to call
+// ImportUsers().
+func (u *UserToImport) PasswordHash(password []byte) *UserToImport {
+	u.userInfo().PasswordHash = base64.RawURLEncoding.EncodeToString(password)
+	return u
+}
+
+// PasswordSalt setter.
+func (u *UserToImport) PasswordSalt(salt []byte) *UserToImport {
+	u.userInfo().Salt = base64.RawURLEncoding.EncodeToString(salt)
+	return u
+}
+
+// ErrorInfo represents an error encountered while importing a single user account.
+//
+// The Index field corresponds to the index of the failed user in the users array that was passed
+// to ImportUsers().
+type ErrorInfo struct {
+	Index  int
+	Reason string
+}
+
+// UserImportResult represents the result of an ImportUsers() call.
+type UserImportResult struct {
```

Comment:
How does this interface compare to other Firebase APIs for batch operations? Does Firestore's batch API return a similar struct?

Reply:
This is similar to https://godoc.org/firebase.google.com/go/messaging#TopicManagementResponse

Firestore seems to return a plain `WriteResult`: https://godoc.org/cloud.google.com/go/firestore#WriteBatch.Commit

Reply:
SGTM